### PR TITLE
Implemented #757. 

### PR DESCRIPTION
**Pull request by koschke**
Previously, all types of configuration files had the same
file extension. The problem is that our backend server needs
to know what to manage based on the file extension.

This PR introduces a distinction of different file extensions
as follows:

- .cfg for code-city configurations (Filenames.CityConfigExtension)
- .drw for drawable configurations (Filenames.DrawableConfigExtension)
- .mbc for metric-board configurations (Filenames.MetricBoardConfigExtension)

Those files use all the same syntax but have different values.

In the course of introducing this distinction, I refactored
SEE.UI.Drawable.DrawableFileBrowser to get rid of some redundancy.
The general selection of path is now moved into a new class
SEE.UI.FilePicker.PathPicker, which may be used elsewhere, too.

I also fixed several bugs in the metric-board code.

In addition, I fixed a few minor coding issues.

### CODE REVIEW

File: Assets/SEE/Game/HolisticMetrics/ActionHelpers/BoardAdder.cs [hunk 1]
```diff
@@ -64,7 +69,12 @@ internal static void Init()
             boardPrefab = Resources.Load<GameObject>("Prefabs/HolisticMetrics/SceneComponents/MetricsBoard");
             positioningDone = false;
             gotPosition = false;
-            GameObject.Find("/DemoWorld/Plane").AddComponent<BoardAdder>();
+            GameObject floor = GameObject.Find(floorPath);
+            if (floor == null)
+            {
+                throw new System.Exception($"Could not find the floor in the scene. Please make sure the floor is named '{floor}'.");
```
Comment (falko17) [inline Assets/SEE/Game/HolisticMetrics/ActionHelpers/BoardAdder.cs:75]: Should probably be `floorPath` instead, as `floor` is null here:
```suggestion
                throw new System.Exception($"Could not find the floor in the scene. Please make sure the floor is at '{floorPath}'.");
```

File: Assets/SEE/UI/Drawable/DrawableFileBrowser.cs [hunk 2]
```diff
@@ -35,160 +32,99 @@ public class DrawableFileBrowser : MonoBehaviour
...
-            if (FileBrowser.Success)
-            {
-                /// Sets the file path upon success.
-                SetPath(FileBrowser.Result[0]);
-            }
-
-            /// Refreshes the UI canvas to prevent display issues.
-            UICanvas.Refresh();
+            PathPicker.GetPath("Load an image", false, initPath, HandleFileBrowserSuccess, () => { },
+                                new string[] { Filenames.PNGExtension, Filenames.JPGExtension });
```
Comment (falko17) [inline Assets/SEE/UI/Drawable/DrawableFileBrowser.cs:127]: Note that JPEG files can have either `.jpg` or `.jpeg` as their file extension, right now we are only catching `.jpg` files.

File: Assets/SEE/UI/FilePicker/PathPicker.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        /// The picked file is returned via the callback <paramref name="onSuccess"/>
+        /// when the user closes the <see cref="FileBrowser"/>.
+        /// If no file has been picked, the callback <paramref name="onCancel"/> is called.
+        /// </summary>
+        /// <param name="title">title of the dialog</param>
+        /// <param name="load">whether the dialog is for loading a file; if false, it is
+        /// for saving a file</param>
+        /// <param name="initialPath">the initial path where the <see cref="FileBrowser"/> allows
+        /// selection/param>
+        /// <paramref name="onSuccess"/> the callback that is called when the user closes
```
Comment (falko17) [inline Assets/SEE/UI/FilePicker/PathPicker.cs:23]:
```suggestion
        /// <paramref name="onSuccess"/>the callback that is called when the user closes
```

File: Assets/SEE/UI/FilePicker/PathPicker.cs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            {
+                SEEInput.KeyboardShortcutsEnabled = true;
+                onCancel();
+            }
+        }
+
+        /// <summary>
+        /// Returns true if the <see cref=">FileBrowser"/> is open.
+        /// </summary>
+        /// <returns>Whether the <see cref=">FileBrowser"/> is open.</returns>
```
Comment (falko17) [inline Assets/SEE/UI/FilePicker/PathPicker.cs:75]:
```suggestion
        /// Returns true if the <see cref="FileBrowser"/> is open.
        /// </summary>
        /// <returns>Whether the <see cref="FileBrowser"/> is open.</returns>
```

**Review by falko17 — CHANGES_REQUESTED** [review]
I have a few minor suggestions.

File: Assets/SEE/UI/Drawable/DrawableFileBrowser.cs (hunk 2 above)
Comment (koschke) [reply Assets/SEE/UI/Drawable/DrawableFileBrowser.cs]: I added these.

**Comment by Cyclone1337** [comment]
All fine. The drawable items are working perfectly. It might be worth considering and evaluating the use of SVG files for the future. The MetricsBoards are working again as well, awesome.

**Review by falko17 — APPROVED** [review]

**Comment by koschke** [comment]
Closes #757

**Pull request merged**